core/src/main/java/io/cucumber/core/runner/CoreHookDefinition.java

```diff
@@ -20,7 +21,13 @@
     private CoreHookDefinition(UUID id, HookDefinition delegate) {
         this.id = requireNonNull(id);
         this.delegate = delegate;
-        this.tagExpression = new TagExpressionParser().parse(delegate.getTagExpression());
+
+        try {
+            this.tagExpression = TagExpressionParser.parse(delegate.getTagExpression());
+        } catch (TagExpressionException tee) {
+            throw new IllegalArgumentException(String.format("%s at '%s'", tee.getMessage(), delegate.getLocation()),
```

Comment:
You originally pushed this down into the different hook implementations. I think because I suggested something about being as close to the source as possible. However this means that we'll have to implement the correct location reporting in all different hook implementations. Using the `getLocation` method we can get the location of the hook. So this is close enough.